invoice/src/main/java/org/killbill/billing/invoice/dao/DefaultInvoiceDao.java

```diff
@@ -395,11 +396,22 @@ public void createInvoice(final InvoiceModelDao invoice,
                                 billingEventSqlDao.create(new InvoiceBillingEventModelDao(invoiceModelDao.getId(), BillingEventSerializer.serialize(billingEvents), context.getCreatedDate()), context);
                             }
                             createdInvoiceIds.add(invoiceModelDao.getId());
-                        } else if ((invoiceOnDisk.getStatus() == InvoiceStatus.DRAFT && invoiceModelDao.getStatus() == InvoiceStatus.COMMITTED) || /* Update if status is needed */
-                                   (invoiceModelDao.getTargetDate() != null && /* Update if target date is specified and prev was either null or prior to new date */
-                                    (invoiceOnDisk.getTargetDate() == null || invoiceOnDisk.getTargetDate().compareTo(invoiceModelDao.getTargetDate()) < 0))) {
-                            invoiceSqlDao.updateStatusAndTargetDate(invoiceModelDao.getId().toString(), invoiceModelDao.getStatus().toString(), invoiceModelDao.getTargetDate(), context);
-                            committedReusedInvoiceId.add(invoiceModelDao.getId());
+                        } else {
+                            // Transition to COMMITTED or keep current status
+                            final InvoiceStatus newStatus = (InvoiceStatus.COMMITTED == invoiceModelDao.getStatus() && InvoiceStatus.COMMITTED != invoiceOnDisk.getStatus()) ? InvoiceStatus.COMMITTED : invoiceOnDisk.getStatus();
```

Comment:
`InvoiceStatus.COMMITTED != invoiceOnDisk.getStatus()` -> `invoiceOnDisk.getStatus() == InvoiceStatus.DRAFT` to be strictly backward compatible (e.g. prevent transition from `VOID` to `COMMITTED`).